[bug] Compilation warnings

The default build command `west build -p -b pinetime hypnos` generates the warnings below. They don't show up when building with bootloader support disabled.

```
pinetime-hypnos/zephyr/include/logging/log_core.h:238:10: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'long unsigned int' [-Wformat=]
  238 |   printk("%c: " fmt "\n", z_log_minimal_level_to_char(_level), \
      |          ^~~~~~
pinetime-hypnos/zephyr/include/logging/log_core.h:266:5: note: in expansion of macro 'Z_LOG_TO_PRINTK'
  266 |     Z_LOG_TO_PRINTK(_level, __VA_ARGS__);        \
      |     ^~~~~~~~~~~~~~~
pinetime-hypnos/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
pinetime-hypnos/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
In file included from pinetime-hypnos/zephyr/include/logging/log.h:11,
                 from pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:22:
pinetime-hypnos/zephyr/include/logging/log_core.h:155:24: warning: format '%x' expects argument of type 'unsigned int', but argument 4 has type 'long unsigned int' [-Wformat=]
  155 | #define Z_LOG_STR(...) "%s: " GET_ARG1(__VA_ARGS__), __func__\
      |                        ^~~~~~
pinetime-hypnos/zephyr/include/logging/log_core.h:188:30: note: in definition of macro '__LOG_INTERNAL'
  188 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
pinetime-hypnos/zephyr/include/logging/log_core.h:279:14: note: in expansion of macro 'Z_LOG_STR'
  279 |              Z_LOG_STR(__VA_ARGS__));\
      |              ^~~~~~~~~
pinetime-hypnos/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
pinetime-hypnos/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
In file included from pinetime-hypnos/zephyr/include/logging/log.h:11,
                 from pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:22:
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:17: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'long unsigned int' [-Wformat=]
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                 ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             |
      |                                                             long unsigned int
pinetime-hypnos/zephyr/include/logging/log_core.h:188:30: note: in definition of macro '__LOG_INTERNAL'
  188 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
pinetime-hypnos/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
pinetime-hypnos/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
In file included from pinetime-hypnos/zephyr/include/logging/log.h:11,
                 from pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:22:
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:17: warning: format '%x' expects argument of type 'unsigned int', but argument 2 has type 'long unsigned int' [-Wformat=]
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                 ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             |
      |                                                             long unsigned int
pinetime-hypnos/zephyr/include/logging/log_core.h:291:27: note: in definition of macro '__LOG'
  291 |    log_printf_arg_checker(__VA_ARGS__);         \
      |                           ^~~~~~~~~~~
pinetime-hypnos/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
pinetime-hypnos/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
```

--- a/app/west.yml
+++ b/app/west.yml
@@ -42,7 +42,8 @@ manifest:
       revision: e88113bbebe34ff2ccc6627ffae885cfeed6fdfd
       path: bootloader/mcuboot
     - name: mcumgr
-      revision: 5885efb7cabf7b566577b73129c9d277d7d8848d
+      url: https://github.com/endian-albin/mcumgr
+      revision: f4e8e00836b25cc2393acb8c14936f7b5356ff2b
       path: modules/lib/mcumgr
     - name: segger
       path: modules/debug/segger

